query/security/ldap/src/main/java/org/codice/ddf/admin/ldap/fields/LdapAttributeName.java

```diff
@@ -28,7 +28,9 @@
     public static final String FIELD_TYPE_NAME = "LdapAttributeName";
 
     public static final String DESCRIPTION =
-            "The name of an attribute that holds information about an LDAP entry.";
+            "The short descriptive name of an LDAP attribute as defined in RFC2251.";
+
+    private static final String REGEX = "^[a-zA-Z0-9]+[a-zA-Z0-9-]*$";
```

Comment:
this should be something like `private static final Pattern ATTRIBUTE_NAME_PATTERN = Pattern.compile("^[a-zA-Z0-9]+[a-zA-Z0-9-]*$")` and then to check do `ATTRIBUTE_NAME_PATTERN.matcher(value).matches()`. Compiling a regex is expensive and we can reduce the amount of times we do it this way.